### PR TITLE
Topic

### PR DESCRIPTION
**Pull request by Adroda**
#### RSLC-1 (https://trello.com/b/Qo2gGF9s/target-adri%C3%A1n)
---
#### Created the topic model and its endpoint
---

### CODE REVIEW

File: db/migrate/20230511192831_create_topics.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateTopics < ActiveRecord::Migration[7.0]
+  def change
+    create_table :topics do |t|
+      t.string :name
+      t.string :image
+
+      t.timestamps
```
Comment (enzofab91) [inline db/migrate/20230511192831_create_topics.rb]: By doing this you are adding this constraint at db level.

```suggestion
      t.string :name, null: false
      t.string :image

      t.timestamps, null: false
```

File: app/models/topic.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+#
+# Table name: topics
+#
+#  id         :bigint           not null, primary key
+#  name       :string
+#  image      :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+class Topic < ApplicationRecord
```
Comment (enzofab91) [inline app/models/topic.rb:11]: What if you add a validation to check `name` presence? Also add it to the spec file.

File: spec/requests/api/v1/topics/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+RSpec.describe 'Topics', type: :request do
+  subject { get api_v1_topics_path, headers: auth_headers, as: :json }
+
+  let(:user) { create(:user) }
+
+  describe 'GET /index' do
+    it 'returns http success' do
+      subject
+      expect(response).to have_http_status(:success)
+    end
```
Comment (enzofab91) [inline spec/requests/api/v1/topics/index_spec.rb]: You can also add a validation checking that topics attributes are being sent as well

**Review by enzofab91 — APPROVED** [review]
Nice work!! Left some comments 💯

File: db/migrate/20230511192831_create_topics.rb (hunk 1 above)
Comment (Adroda) [reply db/migrate/20230511192831_create_topics.rb]: Thanks for the review, its been changed

File: app/models/topic.rb (hunk 2 above)
Comment (Adroda) [reply app/models/topic.rb:11]: Its done, thanks

File: spec/requests/api/v1/topics/index_spec.rb (hunk 3 above)
Comment (Adroda) [reply spec/requests/api/v1/topics/index_spec.rb]: Done

**Pull request merged**